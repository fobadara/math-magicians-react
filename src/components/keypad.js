import PropTypes from 'prop-types';

// Create button tags
const createButtons = (props) => props.keys.map((key) => (
  <button
    name={key.value}
    type="button"
    onClick={props.onClick}
    id={key.id}
    className={key.class}
    key={key.id}
  >
    {key.value}
  </button>
));

const Keypad = (props) => (
  <span id="keypad-container">{createButtons(props)}</span>
);

createButtons.propTypes = {
  props: PropTypes.arrayOf(
    PropTypes.shape({}),
  ).isRequired,
};

export default Keypad;
