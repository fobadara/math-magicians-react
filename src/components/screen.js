/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';

class Screen extends Component {
  render() {
    return (
      <div id="screen" aria-label="screen">0</div>
    );
  }
}

export default Screen;
