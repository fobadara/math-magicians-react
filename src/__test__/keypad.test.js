import {
  render, cleanup, fireEvent, screen,
} from '@testing-library/react';
import Keypad from '../components/keypad';
import { keys } from '../components/calculator';
import '@testing-library/jest-dom/extend-expect';

describe('Keypad', () => {
  afterEach(cleanup);

  test('should create snapshot', () => {
    const { asFragment } = render(<Keypad keys={keys} />);
    expect(asFragment(<Keypad keys={keys} />)).toMatchSnapshot();
  });

  test('should render 19 buttons', () => {
    const { container } = render(<Keypad keys={keys} />);

    expect(container.firstElementChild.children).toHaveLength(19);
  });

  test('should contain correct text (in this case 7)', () => {
    const { container } = render(<Keypad keys={keys} />);

    expect(container.firstElementChild.firstElementChild).toHaveTextContent('AC');
  });

  test('should call onClick when clicked', () => {
    const onClick = jest.fn();
    render(<button name="7" type="button" onClick={onClick} id="7" className="number" key="7">7</button>);
    const number7 = screen.getByText('7');
    fireEvent.click(number7);
    expect(onClick).toHaveBeenCalled();
  });
});
