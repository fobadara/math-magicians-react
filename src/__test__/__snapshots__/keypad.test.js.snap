// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should create snapshot 1`] = `
<DocumentFragment>
  <span
    class="min-vh-75"
    id="keypad-container"
  >
    <button
      class="operators"
      id="clear"
      name="AC"
      type="button"
    >
      AC
    </button>
    <button
      class="numbers"
      id="plus/minus"
      name="+/-"
      type="button"
    >
      +/-
    </button>
    <button
      class="operators"
      id="percent"
      name="%"
      type="button"
    >
      %
    </button>
    <button
      class="operators right"
      id="division"
      name="รท"
      type="button"
    >
      รท
    </button>
    <button
      class="numbers"
      id="seven"
      name="7"
      type="button"
    >
      7
    </button>
    <button
      class="numbers"
      id="eight"
      name="8"
      type="button"
    >
      8
    </button>
    <button
      class="numbers"
      id="nine"
      name="9"
      type="button"
    >
      9
    </button>
    <button
      class="operators right"
      id="multiply"
      name="x"
      type="button"
    >
      x
    </button>
    <button
      class="numbers"
      id="four"
      name="4"
      type="button"
    >
      4
    </button>
    <button
      class="numbers"
      id="five"
      name="5"
      type="button"
    >
      5
    </button>
    <button
      class="numbers"
      id="six"
      name="6"
      type="button"
    >
      6
    </button>
    <button
      class="operators right"
      id="subtract"
      name="-"
      type="button"
    >
      -
    </button>
    <button
      class="numbers"
      id="one"
      name="1"
      type="button"
    >
      1
    </button>
    <button
      class="numbers"
      id="two"
      name="2"
      type="button"
    >
      2
    </button>
    <button
      class="numbers"
      id="three"
      name="3"
      type="button"
    >
      3
    </button>
    <button
      class="operators right"
      id="add"
      name="+"
      type="button"
    >
      +
    </button>
    <button
      class="numbers"
      id="zero"
      name="0"
      type="button"
    >
      0
    </button>
    <button
      class="operators"
      id="decimal"
      name="."
      type="button"
    >
      .
    </button>
    <button
      class="operators right"
      id="equals"
      name="="
      type="button"
    >
      =
    </button>
  </span>
</DocumentFragment>
`;
